app/actions/realm/user.js

```diff
@@ -0,0 +1,208 @@
+
+import {GeneralTypes, UserTypes} from 'app/action_types';
+import {setAppCredentials} from 'app/init/credentials';
+import {GENERAL_SCHEMA_ID} from 'app/models/general';
+import ephemeralStore from 'app/store/ephemeral_store';
+import {configureRealmStore} from 'app/store/';
+import {setCSRFFromCookie} from 'app/utils/security';
+import {getDeviceTimezone} from 'app/utils/timezone';
+
+import {saveConfigAndLicense} from './general';
+import {forceLogoutIfNecessary} from './helpers';
+
+// TODO: Remove redux compatibility
+import {completeLogin} from 'mattermost-redux/actions/users';
+import {reduxStore} from 'app/store';
+import {handleSuccessfulLogin as handleSuccessfulLoginRedux} from 'app/actions/views/login';
+
+export function login(options) {
+    return async () => {
+        const {config, ldapOnly, license, loginId, mfaToken, password} = options;
+        let data = null;
+        try {
+            data = await Client4.login(loginId, password, mfaToken, ephemeralStore.deviceToken, ldapOnly);
+        } catch (error) {
+            return {error};
+        }
+
+        const token = Client4.getToken();
+        const url = Client4.getUrl();
+        const realm = configureRealmStore(url);
+
+        ephemeralStore.currentServerUrl = url;
+        setAppCredentials(ephemeralStore.deviceToken, data.id, token, url);
+        await setCSRFFromCookie(url);
+        await realm.dispatch(saveConfigAndLicense(config, license));
+        await realm.dispatch(loadMe(data));
+        realm.dispatch(handleSuccessfulLogin(config, license));
+
+        // TODO: Remove redux compatibility
+        reduxStore.dispatch(completeLogin(data));
+
+        return data;
+    };
+}
+
+export function ssoLogin(options) {
+    return async () => {
+        const {config, license, token} = options;
+        const url = Client4.getUrl();
+        const realm = configureRealmStore(url);
+
+        Client4.setToken(token);
+        await setCSRFFromCookie(url);
+        await realm.dispatch(saveConfigAndLicense(config, license));
+
+        const data = await realm.dispatch(loadMe());
+        if (data.error) {
+            return data;
+        }
+
+        ephemeralStore.currentServerUrl = url;
+        setAppCredentials(ephemeralStore.deviceToken, data.user.id, token, url);
+        realm.dispatch(handleSuccessfulLogin(config, license));
+
+        // TODO: Remove redux compatibility
+        reduxStore.dispatch(completeLogin(data.user));
+
+        return data;
+    };
+}
+
+export function handleSuccessfulLogin(config, license) {
+    return async (dispatch) => {
+        // TODO: Remove redux compatibility
+        reduxStore.dispatch(handleSuccessfulLoginRedux());
+
+        const enableTimezone = config?.ExperimentalTimezone === 'true';
+        if (enableTimezone) {
+            dispatch(autoUpdateTimezone(getDeviceTimezone()));
+        }
+
+        if (config?.EnableCustomEmoji === 'true') {
+            // TODO: Fetch all custom emojis
```

Comment:
We can remove this since it was only needed pre-4.7, and we don't support that far back any more.